### PR TITLE
Subscription for Individual Events

### PR DESCRIPTION
**Pull request by sericcson**

### CODE REVIEW

File: src/test/java/com/ericsson/ei/subscriptionhandler/test/SubscriptionHandlerTest.java [hunk 1]
```diff
@@ -137,6 +138,20 @@ public void MissedNotificationTest() {
         JsonNode output = jsonResult.get("AggregatedObject");
         assertEquals(expectedOutput.toString(), output.toString());
     }
+
+
+    @Test
+    public void sendMailTest() {
```
Comment (tobiasake) [inline src/test/java/com/ericsson/ei/subscriptionhandler/test/SubscriptionHandlerTest.java]: Can you be more specific in test function name on what you are testing, e.g. "extractEmailsTest()" or something.

File: src/main/java/com/ericsson/ei/mongodbhandler/MongoDBHandler.java [hunk 2]
```diff
@@ -265,12 +277,15 @@ public void createTTLIndex(String dataBaseName, String collectionName, String fi
     }
 
     private MongoCollection<Document> getMongoCollection(String dataBaseName, String collectionName) {
+        if (mongoClient == null)
+            return null;
         MongoDatabase db = mongoClient.getDatabase(dataBaseName);
-        if (!db.listCollectionNames().into(new ArrayList<String>()).contains(collectionName)) {
-            log.info("The requested database(" + dataBaseName + ") / collection(" + collectionName
+        ArrayList<String> collectionList = db.listCollectionNames().into(new ArrayList<String>());
```
Comment (ned29) [inline src/main/java/com/ericsson/ei/mongodbhandler/MongoDBHandler.java]: I think should be List

File: src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.when;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest
+public class SingleEventAggregationTest extends FlowTestBase {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FlowTest.class);
+
+    @Autowired
+    RulesHandler rulesHandler;
```
Comment (ned29) [inline src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java]: make it private

File: src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    public ObjectHandler objectHandler;
+
+    @Autowired
+    private UpStreamEventsHandler upStreamEventsHandler;
+
+    @Mock
+    private ERQueryService erQueryService;
+
+    private static final String RULES_FILE_PATH = "src/test/resources/all_event_rules.json";
+    private static final String EVENTS_FILE_PATH = "src/test/resources/test_All_Events.json";
```
Comment (ned29) [inline src/test/java/com/ericsson/ei/flowtests/SingleEventAggregationTest.java]: move it to top of class, code convention

File: src/main/java/com/ericsson/ei/mongodbhandler/MongoDBHandler.java (hunk 2 above)
Comment (sericcson) [reply src/main/java/com/ericsson/ei/mongodbhandler/MongoDBHandler.java]: True...OK

**Pull request merged**